validation/gas_token.go

```diff
@@ -0,0 +1,104 @@
+package validation
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/stretchr/testify/require"
+)
+
+func testGasToken(t *testing.T, chain *superchain.ChainConfig) {
+	l1Client, err := ethclient.Dial(superchain.Superchains[chain.Superchain].Config.L1.PublicRPC)
+	require.NoError(t, err, "Failed to connect to the Ethereum client at RPC url %s", chain.PublicRPC)
+	defer l1Client.Close()
+
+	err = CheckGasToken(chain, l1Client)
+	require.NoError(t, err)
+}
+
+func CheckGasToken(chain *superchain.ChainConfig, l1Client EthClient) error {
+	l2Client, err := ethclient.Dial(chain.PublicRPC)
```

Comment:
Shouldn't we pass in a l2Client so that we can mock it?

Reply:
Yea will move the l2Client connection into the test wrapper then pass it in here